libraries/AP_SmartAudio/AP_SmartAudio.h

```diff
@@ -0,0 +1,205 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Vehicle/AP_Vehicle_Type.h>
+#include <AP_Param/AP_Param.h>
+#include <stdio.h>
+
+#ifndef SMARTAUDIO_ENABLED
+#define SMARTAUDIO_ENABLED !HAL_MINIMIZE_FEATURES && !APM_BUILD_TYPE(APM_BUILD_Replay)
+#endif
+
+#if SMARTAUDIO_ENABLED
+#define SMARTAUDIO_BUFFER_CAPACITY 5
+
+#include "smartaudio_protocol.h"
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <AP_HAL/utility/RingBuffer.h>
+#include <AP_RCTelemetry/AP_VideoTX.h>
+
+
+class AP_SmartAudio
+{
+
+public:
+
+    // request packet to be processed
+    struct packet{
+        smartaudioFrame_t frame;
+        uint8_t frame_size=0;
+        uint32_t sended_at=0;
+    } PACKED;
+
+    // configured external params
+    struct configured_smartaudio_params{
+        AP_Int8  enabled;
+        AP_Int8  protocol_version;
+        AP_Int8  setup_defaults;
+    } smartaudio_params;
+
+    AP_SmartAudio();
+
+    static AP_SmartAudio *get_singleton(void)
+    {
+        return singleton;
+    }
+
+    /* Do not allow copies */
+    AP_SmartAudio(const AP_SmartAudio &other) = delete;
+
+    AP_SmartAudio &operator=(const AP_SmartAudio&) = delete;
+
+    static const struct AP_Param::GroupInfo var_info[];
+
+    // init threads and lookup for io uart.
+    bool init();
+
+    // looping over request loop
+    void loop();
+
+    // updates the smartaudio state in sync whith AP_VideoTX
+    bool update(bool force);
+
+    // RingBuffer to store outgoing request.
+    ObjectBuffer<packet> requests_queue{SMARTAUDIO_BUFFER_CAPACITY};
+
+    // time the last_request is process
+    uint32_t last_request_sended_at=0;
+    // loops is waiting a response after a request
+    bool is_waiting_response=false;
+
+    // sends a frame over the wire
+    void send_request(smartaudioFrame_t requestFrame,uint8_t size);
+
+    // receives a frame response over the wire
+    void read_response(uint8_t *response_buffer,uint8_t inline_buffer_length);
+
+    // parses the response and updates the AP_VTX readings
+    bool parse_frame_response(const uint8_t *buffer);
+
+    // to prevent use of semaphores use an vtx_states_queue with 2 elements, so one buffered reading
+    ObjectBuffer<smartaudioSettings_s> vtx_states_queue{2};
+
+    // get last reading from the fifo queue
+    bool get_readings(AP_VideoTX *vtx_dest);
+
+    // enqueue a get settings request
+    void request_settings();
+
+    void set_operation_mode(uint8_t mode);
+
+    // enqueue a set frecuency request, specifiying if the setted frequency is for pit mode or not
+    void set_frequency(uint16_t frecuency,bool isPitModeFreq);
+
+    // enqueue a set channel request
+    void set_channel(uint8_t chan);
+
+    // enqueue a get pit mode frequency request
+    void get_pit_mode_frequency();
```

Comment:
You are not returning anything so the name does not make sense

Reply:
Rename to request_pit_mode_frequency()